distributed/diagnostics/tests/test_cluster_dump_plugin.py

```diff
@@ -0,0 +1,18 @@
+from distributed.diagnostics.cluster_dump import ClusterDump
+from distributed.utils_test import gen_cluster, inc
+
+
+@gen_cluster(client=True)
+async def test_cluster_dump_plugin(c, s, *workers, tmp_path):
+    dump_file = tmp_path / "cluster_dump.msgpack.gz"
+
+    plugin = ClusterDump(s, str(dump_file), [], "msgpack")
+    s.add_plugin(plugin)
+
+    f1 = c.submit(inc, 1)
+    f2 = c.submit(inc, f1)
+
+    assert (await f2) == 3
+    await s.close(close_workers=True)
+
+    assert dump_file.exists()
```

Comment:
_maybe_ test that `f1` and `f2` are in known keys, to confirm the dump wasn't written until the end? Like `DumpArtefact.from_url(dump_file).scheduler_tasks_in_state("memory") == [...]`?

Reply:
I've added the following to the test case:

```python
dump = DumpArtefact.from_url(str(dump_file))
assert {f1.key, f2.key} == set(dump.scheduler_story(f1.key, f2.key).keys())
assert {f1.key, f2.key} == set(dump.worker_story(f1.key, f2.key).keys())
```